### PR TITLE
[Certicates] Remove certificate migration

### PR DESCRIPTION
**Pull request by legionth**
As mentioned on the last ILIAS Developers the migration of certificates will be removed for upcoming releases.

This PR will remove the code that contains this migration service.

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
Thx @legionth for this PR.

Please implement the following changes:
- [x] Please add a database update step which cleans up related `Certificate Job` data in the `BackgroundTask` database tables.
- [x] Please add a database update step which removes table `il_cert_bgtask_migr`.
- [x] The changes applied to `ilCertificateAppEventListener` seem to be incomplete.

File: Services/LearningHistory/classes/class.ilLearningHistoryTimelineItem.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function render()
 		$f = $this->ui->factory();
 		$r = $this->ui->renderer();
 
-		$ico = $f->symbol()->icon()->custom($this->lh_entry->getIconPath(), '')->withSize(\ILIAS\UI\Component\Icon\Custom::MEDIUM);
+		$ico = $f->symbol()->icon()->custom($this->lh_entry->getIconPath(), '')->withSize(\ILIAS\UI\Component\Symbol\Icon\Icon::MEDIUM);
```
Comment (mjansenDatabay) [inline Services/LearningHistory/classes/class.ilLearningHistoryTimelineItem.php]: ?

File: setup/sql/dbupdate_05.php [hunk 2]
```diff
@@ -1067,4 +1067,9 @@
 	$ilDB->dropTable($tempTableName);
 }
 
-?>
\ No newline at end of file
+?>
+<#5503>
+<?php
+if(!$ilDB->tableExists('bgtask_cert_migration')) {
```
Comment (mjansenDatabay) [inline setup/sql/dbupdate_05.php]: `if ($ilDB->tableExists('bgtask_cert_migration')) {`

File: Services/LearningHistory/classes/class.ilLearningHistoryTimelineItem.php (hunk 1 above)
Comment (legionth) [reply Services/LearningHistory/classes/class.ilLearningHistoryTimelineItem.php]: This was based on an old trunk branch, before #1890 was merged.

File: setup/sql/dbupdate_05.php [hunk 3]
```diff
@@ -1099,3 +1099,26 @@
 if(!$ilDB->tableExists('il_cert_bgtask_migr')) {
 	$ilDB->dropTable('il_cert_bgtask_migr');
 }
+?>
+<#5505>
+<?php
+if ($ilDB->tableExists('il_bt_value')) {
+    $deleteBucketsSql = 'DELETE FROM il_bt_value WHERE bucket_id IN (SELECT id FROM il_bt_bucket WHERE title = "Certificate Migration")';
```
Comment (mjansenDatabay) [inline setup/sql/dbupdate_05.php]: Please use `ilDBInterface::quote()` here and below.

File: setup/sql/dbupdate_05.php [hunk 4]
```diff
@@ -1099,3 +1099,26 @@
...
+    $ilDB->manipulate($deleteBucketsSql);
+}
+
+if ($ilDB->tableExists('il_bt_value_to_task')) {
+    $deleteValueToTask = 'DELETE FROM il_bt_value_to_task WHERE bucket_id IN (SELECT id FROM il_bt_bucket WHERE title = "Certificate Migration")';
```
Comment (mjansenDatabay) [inline setup/sql/dbupdate_05.php]: Please use `ilDBInterface::in()`

File: setup/sql/dbupdate_05.php [hunk 5]
```diff
@@ -1103,22 +1103,22 @@
...
     $ilDB->manipulate($deleteBucketsSql);
 }
 
 if ($ilDB->tableExists('il_bt_value_to_task')) {
-    $deleteValueToTask = 'DELETE FROM il_bt_value_to_task WHERE bucket_id IN (SELECT id FROM il_bt_bucket WHERE title = "Certificate Migration")';
+    $deleteValueToTask = 'DELETE FROM il_bt_value_to_task WHERE bucket_id IN (SELECT id FROM il_bt_bucket WHERE title =  ' . $ilDB->quote('Certificate Migration', 'text') . ')';
     $ilDB->manipulate($deleteValueToTask);
 }
 
 if ($ilDB->tableExists('il_bt_task')) {
```
Comment (mjansenDatabay) [inline setup/sql/dbupdate_05.php:1105]: I am not sure if `ilCertificateMigrationReducedInteraction` is sufficient here. We also had `ilCertificateMigrationInteraction` and `ilCertificateMigrationJob`.

Comment (legionth) [reply setup/sql/dbupdate_05.php:1105]: `LIKE "ilCertificateMigration%"`?

File: setup/sql/dbupdate_05.php [hunk 6]
```diff
@@ -1093,4 +1093,42 @@
...
+)';
+
+        $ilDB->manipulate($deleteValueToTask);
+    }
+    $deleteBackgroundTasksSql = 'DELETE FROM il_bt_task WHERE ' . $ilDB->like('type', 'text', 'ilCertificateMigration%');
+    $ilDB->manipulate($deleteBackgroundTasksSql);
+}
+
+if ($ilDB->tableExists('il_bt_bucket')) {
+    $deleteBucketsSql = 'DELETE FROM il_bt_bucket WHERE title = ' . $ilDB->quote('Certificate Migration', 'text') ;
```
Comment (mjansenDatabay) [inline setup/sql/dbupdate_05.php:1130]: Do we need a LIKE here as well?

**Pull request merged**